mentoring/mentoring.py

```diff
@@ -83,54 +65,105 @@ class MentoringBlock(XBlockWithLightChildren, StepParentMixin):
-    followed_by = String(help="url_name of the step after the current mentoring block in workflow",
-                         default=None, scope=Scope.content)
-    url_name = String(help="Name of the current step, used for URL building",
-                      default='mentoring-default', scope=Scope.content)
-    enforce_dependency = Boolean(help="Should the next step be the current block to complete?",
-                                 default=False, scope=Scope.content, enforce_type=True)
+        return value == default_xml_content
+
+    # Content
+    MENTORING_MODES = ('standard', 'assessment')
+    mode = String(
+        help="Mode of the mentoring. 'standard' or 'assessment'",
+        default='standard',
+        scope=Scope.content,
+        values=MENTORING_MODES
+    )
+    followed_by = String(
+        help="url_name of the step after the current mentoring block in workflow.",
+        default=None,
+        scope=Scope.content
+    )
+    max_attempts = Integer(
+        help="Number of max attempts for this questions",
+        default=0,
+        scope=Scope.content,
+        enforce_type=True
+    )
+    url_name = String(
+        help="Name of the current step, used for URL building",
+        default='mentoring-default',  # TODO in future: set this to xblock.fields.UNIQUE_ID and remove self.url_name_with_default
+        scope=Scope.content
+    )
+    enforce_dependency = Boolean(
+        help="Should the next step be the current block to complete?",
+        default=False,
+        scope=Scope.content,
+        enforce_type=True
+    )
     display_submit = Boolean(help="Allow to submit current block?", default=True, scope=Scope.content)
-    xml_content = String(help="XML content", default=_default_xml_content, scope=Scope.content)
-    weight = Float(help="Defines the maximum total grade of the block.",
-                   default=1, scope=Scope.content, enforce_type=True)
-    num_attempts = Integer(help="Number of attempts a user has answered for this questions",
-                           default=0, scope=Scope.user_state, enforce_type=True)
-    max_attempts = Integer(help="Number of max attempts for this questions", default=0,
-                           scope=Scope.content, enforce_type=True)
-    mode = String(help="Mode of the mentoring. 'standard' or 'assessment'",
-                  default='standard', scope=Scope.content)
-    step = Integer(help="Keep track of the student assessment progress.",
-                   default=0, scope=Scope.user_state, enforce_type=True)
-    student_results = List(help="Store results of student choices.", default=[],
-                           scope=Scope.user_state)
-
-    display_name = String(help="Display name of the component", default="Mentoring XBlock",
-                          scope=Scope.settings)
+    xml_content = String(help="XML content", default=default_xml_content, scope=Scope.content)
```

Comment:
@bradenmacdonald @e-kolpakov It's hard to be sure that we aren't forgetting edge cases with these changes, but greping the code I could only see this use having an impact:

https://github.com/edx-solutions/xblock-mentoring/blob/a8838b80f65a4d2066f523a31bdca69f05c89ad5/mentoring/mentoring.py#L193

With a non-random url_name it means the component_id logged in events would be the same for all emntoring newly created XBlocks for which this isn't edited. We might be able to get away with using another unique id there if the url_name is the default one when an event is emitted.

But we should really be using the feature we specifically contributed for this use case: https://github.com/edx/XBlock/pull/249 If the XBlock repo is out of sync with edx-platform, if might be worth pushing a PR on edx-platform for this, and ask if this can into birch.